web-app/pages/api/invalidate-all-certs-for-account.ts

```diff
@@ -6,36 +6,46 @@ import { depositAmountYoctoNear } from './mint-cert';
 const apiKeyHeaderName = 'x-api-key'; // Although the user interface of Integromat shows the capitalization as "X-API-Key", inspecting the actual header reveals that lowercase is used.
 
 async function invalidateAllCertsForAccount(accountId: AccountId) {
-  return;
-  // TODO: This function isn't implemented yet. Is there a way to await each cert_invalidate and then return a result once all have succeeded?
   const contract = await getNftContract();
   const tokens = await (contract as NFT).nft_tokens_for_owner(accountId);
-  tokens.forEach((token: string) => {
-    (contract as NFT).cert_invalidate(
-      // https://github.com/near/near-api-js/issues/719
-      token,
-      gas, // attached GAS (optional)
-      depositAmountYoctoNear, // attached deposit in yoctoNEAR (optional).
+
+  try {
+    await Promise.all(
+      tokens.map((token: string) =>
+        (contract as NFT).cert_invalidate(
+          // https://github.com/near/near-api-js/issues/719
+          token,
+          gas, // attached GAS (optional)
+          depositAmountYoctoNear, // attached deposit in yoctoNEAR (optional).
+        ),
+      ),
     );
-  });
+
+    return { success: true };
```

Comment:
@encody You'll be a 3rd set of eyes on this. Are these lines good? invalidateAllCertsForAccount should loop through and call cert_invalidate for every NFT found associated with the given account.  I haven't done much with NEAR async so want to check.

Reply:
@encody This also ties in with the [signAndSendTransaction](https://github.com/near/near-api-js/blob/6f83d39f47624b4223746c0d27d10f78471575f7/src/account.ts#L217) being `protected` even though there is an example in the official docs cookbook [here](https://docs.near.org/docs/api/naj-cookbook#batch-transactions). I would prefer to have done it in one transaction (which would decrease cost and time) but I settled for this for now. Don't know how to avoid it, (besides manually writing the `signAndSendTransaction` function for our usacase)

Reply:
With some slight modification, I got it working. Please review and test :)